### ISSUE TITLE
Stricter CI for CSS (stylelint), JS (eslint)

### ISSUE THREAD

**Issue by mattxwang** [uclaacm/website#114]
As our codebase becomes larger and changes hands more frequently, automated style enforcing will go a long way. We can do that with **continuous integration**, where we run certain tests on our code every commit/push.

In other words, if you like automating things, this issue is for you!

The simplest CI integrations we can do (other than building, which we already do) are linting: static code analysis to make sure we don't have errors, avoid bad practice, and maintain a consistent style across the entire repository.

There are two tools we can look to loop in for now:

- [ ] [stylelint](https://stylelint.io/), a highly-configurable CSS linter
- [x] [eslint](https://eslint.org/), the defacto-standard JS linter (partially built in already)

I already wrote up a similar guide on how to integrate both of these into our workflow [for Teach LA](https://github.com/uclaacm/tla-dev-intern-training-w21/tree/main/week-6d-ci-cd-tests#cicd-in-practice-linters), so I encourage you to read through that; it should relatively walk you through how to set these up.

For posterity's sake, I suggest you:

0. make a new branch,  etc.
1. install and configure stylelint; make sure that you can run it! **do not run the autofixer yet**
2. configure an action to run stylelint on every commit. **do not run the auto fixer yet; it's okay if the action fails**
3. open up a draft PR and request a review
4. once that's approved, then, run the autofixer
5. wrap up the PR and merge it 
6. rinse and repeat 0-5 for eslint

Some things I'll note:

* for now, we can use a lax JSX/js styleguide. [Bryan from Creative Labs](https://github.com/uclaacm/website/issues/67#issuecomment-814538707) recommended [Creative Labs' config](https://github.com/UCLA-Creative-Labs/sunshine/blob/master/.eslintrc.js), which I quite enjoy. 
* eslint is already part of Create React App, so you won't need to do *as much* setup as stylelint
* standard config for stylelint is pretty good imo, you won't need to make any tweaks
* we use SASS (Syntactically Awesome Style Sheets), so you may want to install a sass plugin for stylelint as well; I included this in the [writeup](https://github.com/uclaacm/tla-dev-intern-training-w21/tree/main/week-6d-ci-cd-tests#stylelint-addendum-sass)

Implementing this then paves the way for us to add some pre-commit hooks, to automatically run the fixer - but that's for another issue 😉 . In parallel (or afterwards), we can apply this same approach for accessibility, link-checking, etc.

Supersedes #67.

**Comment by mattxwang** [comment]
As an update, you now only need to do stylelint - I addressed the JS stuff in #120!

**Comment by mattxwang** [comment]
Closed (for now) by #150!
